remove non-test dependency on testutils

RaduBerinde opened the pull request:
`LocalTestCluster` has a dependency on `testutils` which is easily removed by
having the caller pass a context. This is testing code but is part of the kv
package because it is used from a test in another package. The dependency
blocked the addition of new functionality that involves `testing`.

Adding a dummy dependency on `testing` so this doesn't happen in the future.

<!-- Reviewable:start -->

---

This change is [<img src="https://reviewable.io/review_button.svg" height="35" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/6459)

<!-- Reviewable:end -->

bdarnell commented:
LGTM (See also #6246)

Instead of adding a dummy dependency, should we add testutils to the blacklist in TestNoLinkForbidden (cli/flags_test.go)?

RaduBerinde committed:
remove dependency on storageutils

RaduBerinde committed:
forbid linking of everything under testutils

Adding TestNoLinkForbidden test for everything under `testutils`. Removing the
dummy testing dependencies which were a roundabout way of doing this.

RaduBerinde commented:
TFTR! Good idea, switched to checking in `TestNoLinkForbidden`. Also removed dependency on `storageutils` so we can forbid everything under `testutils`.

---

Review status: 0 of 21 files reviewed at latest revision, all discussions resolved, some commit checks failed.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/6459#-:-KGqZu7DO4Tlnd7KHhJT:-1193796060)*
<!-- Sent from Reviewable.io -->

bdarnell commented:
Reviewed 4 of 6 files at r1, 16 of 16 files at r2, 3 of 3 files at r3.
Review status: all files reviewed at latest revision, all discussions resolved, some commit checks failed.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/6459)*
<!-- Sent from Reviewable.io -->

RaduBerinde committed:
remove non-test dependency on testutils

`LocalTestCluster` has a dependency on `testutils` which is easily removed by
having the caller pass a context. This is testing code but is part of the kv
package because it is used from a test in another package. The dependency
blocked the addition of new functionality that involves `testing`.

Adding a dummy dependency on `testing` so this doesn't happen in the future.
